scrunch/scripts.py

```diff
@@ -44,19 +46,50 @@ def get_default_syntax_flag(self, strict_subvariable_syntax):
         """
         if strict_subvariable_syntax is not None:
             return strict_subvariable_syntax
-        flags = self.dataset_resource.session.feature_flags
+        flags = self.resource.session.feature_flags
         return flags.get("clients_strict_subvariable_syntax", DEFAULT_SUBVARIABLE_SYNTAX)
 
+    def execute(self, script_body, strict_subvariable_syntax=None):
+        pass
+
+
+class SystemScript(BaseScript):
+
+    def format_request_url(self, execute, strict_subvariable_syntax=None):
+        strict_subvariable_syntax_flag = self.get_default_syntax_flag(strict_subvariable_syntax)
+        if strict_subvariable_syntax_flag:
+            execute.self += "?strict_subvariable_syntax=true"
```

Comment:
Here you will be mutating the `self` attribute of the `execute` resource. I think that this better returns the desired URL instead of altering the instance.

Because when you execute a script for the 2nd time this will add again this string incorrectly

Reply:
That is not completely true. I am not mutating the self.resource instance, but just the local `execute`.
Every time you do `self.resource.something` there is a GET call to the `resource` o get the attribute. 
So the self.resource instance is not modified by this, I am just updating the local object that is the result of the GET
So here https://github.com/Crunch-io/scrunch/blob/86b3b7337d498c057c8f1af443d5aa056cc1073e/scrunch/scripts.py#L73-L73 I am fetching the `execute` object, and I am modifing this one that is the result of this call https://github.com/Crunch-io/pycrunch/blob/9bb073368293f1e84843a53bdc0cfccd1c90787e/src/pycrunch/elements.py#L154
Next time, next script, I am going to redo the request to the same self.resource object, that has not being modified.
Let me know if that works for you. Otherwise I'll think of another way of doing this. But I just wanted to use the pycrunch interface as per the DatasetsScripts